src/plugins/profiling/server/routes/search_flamechart.ts

```diff
@@ -172,61 +173,70 @@ async function queryFlameGraph(
   // profiling-stacktraces is configured with 16 shards
   const nQueries = 8;
   const stackTraces = new Map<StackTraceID, StackTrace>();
+  const stackFrameDocIDs = new Set<string>(); // Set of unique FrameIDs
+  const executableDocIDs = new Set<string>(); // Set of unique executable FileIDs.
+  const stackTraceIDs = [...stackTraceEvents.keys()];
+  const chunkSize = Math.floor(stackTraceEvents.size / nQueries);
+  const chunks = chunk(stackTraceIDs, chunkSize);
 
-  await logExecutionLatency(
+  const stackResponses = await logExecutionLatency(
     logger,
     'mget query for ' + stackTraceEvents.size + ' stacktraces',
     async () => {
-      const promises = new Array(nQueries);
-      const chunkSize = Math.floor(stackTraceEvents.size / nQueries);
-      const stackTraceIDs = [...stackTraceEvents.keys()];
-
-      for (let i = 0; i < nQueries; i++) {
-        const func = async () => {
-          const chunk = stackTraceIDs.slice(chunkSize * i, chunkSize * (i + 1));
-          const res = await client.mget({
+      return await Promise.all(
+        chunks.map((ids) => {
+          return client.mget({
             index: 'profiling-stacktraces',
-            ids: [...chunk],
+            ids,
             _source_includes: ['FrameID', 'Type'],
           });
+        })
+      );
+    }
+  );
 
-          if (testing) {
-            for (const trace of res.body.hits.hits) {
-              const frameIDs = trace.fields.FrameID as string[];
-              const fileIDs = extractFileIDArrayFromFrameIDArray(frameIDs);
-              stackTraces.set(trace._id, {
-                FileID: fileIDs,
-                FrameID: frameIDs,
-                Type: trace.fields.Type,
-              });
-            }
-          } else {
-            for (const trace of res.body.docs) {
-              // Sometimes we don't find the trace.
-              // This is due to ES delays writing (data is not immediately seen after write).
-              // Also, ES doesn't know about transactions.
-              if (trace.found) {
-                const frameIDs = trace._source.FrameID as string[];
-                const fileIDs = extractFileIDArrayFromFrameIDArray(frameIDs);
-                stackTraces.set(trace._id, {
-                  FileID: fileIDs,
-                  FrameID: frameIDs,
-                  Type: trace._source.Type,
-                });
-              }
-            }
+  await logExecutionLatency(logger, 'processing data', async () => {
+    if (testing) {
+      const traces = stackResponses.flatMap((response) => response.body.hits.hits);
+      for (const trace of traces) {
```

Comment:
Is it possible to condense the for loop and the previous iterator into a single one?